tests/testing_utils/fixtures.py

```diff
@@ -338,3 +338,54 @@ def native_app_project_instance():
             },
         }
     )
+
+
+@pytest.fixture()
+def native_app_codegen_instance():
+    return {
+        "object_type": "TABLE_FUNCTION",
+        "object_name": "SNOWPARK_TEMP_FUNCTION_WZUNHMZJKA",
```

Comment:
for now I would prefer to focus on the case where the name was explicitly provided, and even make that an strong requirement for the users. The other stuff we do is really just a heuristic. We could deliver without, if need be.

Reply:
I don't believe we can distinguish between Snowpark's temp name and a user-provided name that happens to begin with this "SNOWPARK_TEMP_" prefix. We do provide precedence to objects that do not begin with "SNOWPARK_TEMP_" prefix. Worst case where a user provided this prefix, we will add the schema name to the rest of the object_name.

Reply:
We should probably make another change to the Snowpark lib to optionally avoid generating names. We need to be able to tell when that's happening, and a prefix isn't the right way. It's too brittle. That being said, we can have the logic to handle the prefix specially for now, but we should really tackle the real solution before that logic is forced to live forever.